test/CAPI/smtlib.c

```diff
@@ -43,10 +43,77 @@ void testExportSMTLIB(MlirContext ctx) {
+      ctx, 2, (MlirType[]){smtTypeGetInt(ctx), smtTypeGetBool(ctx)},
+      smtTypeGetBool(ctx))));
+  assert(smtTypeIsASort(smtTypeGetSort(
+      ctx, mlirIdentifierGet(ctx, mlirStringRefCreateFromCString("sort")), 0,
+      NULL)));
+
+  assert(smtTypeIsAnyNonFuncSMTValueType(smtTypeGetBool(ctx)));
+  assert(smtTypeIsAnyNonFuncSMTValueType(smtTypeGetInt(ctx)));
+  assert(smtTypeIsAnyNonFuncSMTValueType(
+      smtTypeGetArray(ctx, smtTypeGetInt(ctx), smtTypeGetBool(ctx))));
+  assert(smtTypeIsAnyNonFuncSMTValueType(smtTypeGetBitVector(ctx, 32)));
+  assert(smtTypeIsAnyNonFuncSMTValueType(smtTypeGetSort(
+      ctx, mlirIdentifierGet(ctx, mlirStringRefCreateFromCString("sort")), 0,
+      NULL)));
+  assert(!smtTypeIsAnyNonFuncSMTValueType(smtTypeGetSMTFunc(
+      ctx, 2, (MlirType[]){smtTypeGetInt(ctx), smtTypeGetBool(ctx)},
+      smtTypeGetBool(ctx))));
+
+  assert(smtTypeIsAnySMTValueType(smtTypeGetBool(ctx)));
+  assert(smtTypeIsAnySMTValueType(smtTypeGetInt(ctx)));
+  assert(smtTypeIsAnySMTValueType(
+      smtTypeGetArray(ctx, smtTypeGetInt(ctx), smtTypeGetBool(ctx))));
+  assert(smtTypeIsAnySMTValueType(smtTypeGetBitVector(ctx, 32)));
+  assert(smtTypeIsAnySMTValueType(smtTypeGetSMTFunc(
+      ctx, 2, (MlirType[]){smtTypeGetInt(ctx), smtTypeGetBool(ctx)},
+      smtTypeGetBool(ctx))));
+  assert(smtTypeIsAnySMTValueType(smtTypeGetSort(
+      ctx, mlirIdentifierGet(ctx, mlirStringRefCreateFromCString("sort")), 0,
+      NULL)));
+
+  assert(!smtTypeIsABool(smtTypeGetInt(ctx)));
+  assert(!smtTypeIsAInt(smtTypeGetBool(ctx)));
+  assert(!smtTypeIsAArray(smtTypeGetBitVector(ctx, 32)));
+  assert(!smtTypeIsABitVector(
+      smtTypeGetArray(ctx, smtTypeGetInt(ctx), smtTypeGetBool(ctx))));
+  assert(!smtTypeIsASMTFunc(smtTypeGetSort(
+      ctx, mlirIdentifierGet(ctx, mlirStringRefCreateFromCString("sort")), 0,
+      NULL)));
+  assert(!smtTypeIsASort(smtTypeGetSMTFunc(
+      ctx, 2, (MlirType[]){smtTypeGetInt(ctx), smtTypeGetBool(ctx)},
+      smtTypeGetBool(ctx))));
+}
+
+void testSMTAttribute(MlirContext ctx) {
+  assert(
+      smtAttrCheckBVCmpPredicate(ctx, mlirStringRefCreateFromCString("slt")));
+  assert(
+      !smtAttrCheckBVCmpPredicate(ctx, mlirStringRefCreateFromCString("xxx")));
+  assert(smtAttrCheckIntPredicate(ctx, mlirStringRefCreateFromCString("lt")));
+  assert(!smtAttrCheckIntPredicate(ctx, mlirStringRefCreateFromCString("xx")));
+
+  assert(smtAttrIsASMTAttribute(smtAttrGetBitVector(ctx, 5, 32)));
+  assert(smtAttrIsASMTAttribute(
+      smtAttrGetBVCmpPredicate(ctx, mlirStringRefCreateFromCString("slt"))));
+  assert(smtAttrIsASMTAttribute(
+      smtAttrGetIntPredicate(ctx, mlirStringRefCreateFromCString("lt"))));
+}
+
 int main(void) {
   MlirContext ctx = mlirContextCreate();
   mlirDialectHandleLoadDialect(mlirGetDialectHandle__smt__(), ctx);
   mlirDialectHandleLoadDialect(mlirGetDialectHandle__func__(), ctx);
   testExportSMTLIB(ctx);
+  testSMTType(ctx);
+  testSMTAttribute(ctx);
   return 0;
 }
```

Comment:
There's a newline missing at the end of the file. Not introduced by this PR, but maybe you can add it?